src/hooks/useUsers.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+import { User } from '../types';
+import * as usersService from '../api/users';
+
+export const useUsers = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const loadUsers = () => {
+    setErrorMessage('');
+
+    usersService
+      .getUsers()
+      .then(res => {
+        const preparedUsers = res.map(user => {
+          const { id, name, email, phone } = user;
+
+          return {
+            id,
+            name,
+            email,
+            phone,
+          };
+        });
```

Comment:
The `preparedUsers` mapping seems redundant if it only extracts and returns the same properties without modification. Consider simplifying this by directly setting `setUsers(res)` unless there's a specific reason for this transformation.